File: ts-sdk/src/scanning/evm/get.block.time.ts
// returns the block time in seconds given a chainId
export type GetBlockTime = (chainId: number) => number;

export function provideGetBlockTime(): GetBlockTime {
  return function getBlockTime(chainId: number) {
    switch (chainId) {
      case 1: // eth
        return 12;
      case 56: // bsc
        return 3;
      default:
        return 2;
    }
  };
}

File: py-sdk/src/forta_bot/scanning/evm/get_block_time.py
from typing import Callable

# returns block time in seconds given a chain id
GetBlockTime = Callable[[int], int]


def provide_get_block_time() -> GetBlockTime:
    def get_block_time(chain_id: int):
        match chain_id:
            case 1:  # eth
                return 12
            case 56:  # bsc
                return 3
            case _:
                return 2

    return get_block_time
